lib/ansible/plugins/action/network.py

```diff
@@ -60,7 +60,12 @@ def _handle_backup_option(self, result, hostname, backup):
             # User requested backup and no error occurred in module.
             # NOTE: If there is a parameter error, _backup key may not be in results.
             filepath = self._write_backup(hostname, backup)
+
             result['backup_path'] = filepath
+            result['filename'] = filepath.split('/')[-1]
+            result['shortname'] = filepath[::-1].split('.', 1)[1][::-1]
+            result['date'] = time.strftime('%Y-%m-%d', time.gmtime(os.stat(filepath).st_ctime))
```

Comment:
The issue with this is the timestamp in the filename and the actual timestamp for file changed might not be the same. Thus returning the wrong timestamp within the filename. To avoid this you can check if configurable backup path is given as input to the module. Refer https://github.com/ansible/ansible/pull/50801

Reply:
Ack. Fixed in the new commit.